### DISCUSSION TITLE
Passing session into component for testing

### DISCUSSION THREAD

**Discussion by danvoyce** [nextauthjs/next-auth#7251] [Help]
### Question 💬

I'm trying to setup some tests (using Jest and testing-library) for my NextJS app. I'm using MSW to mock all my apis, including the session, however I need pass `session` into the `SessionProvider` somehow.

### How to reproduce ☕️

My MSW handler looks like this:

```ts
...
rest.get("/api/auth/session", (req, res, ctx) => {
    return res(
      ctx.json({
        user: {
          name: "Jim Doe",
        },
        accessToken: "123",
      })
    );
  }),
...
```

and I'm using the `withAuth` middleware, so everything works nicely in the browser.

However within the tests I must essentially recreate my `_app.tsx` file in order to wrap my `Home` page with the `SessionProvider`:

```tsx
function setup() {
  render(
    <SessionProvider session={"???"}>
          <Home />
    </SessionProvider>
  );
}
```

I can pass in a mock session and all is good, however it would be nice to use the "real" one in the MSW handler, which would usually be available as a prop within `_app.tsx`.

Thanks in advance for any help!

### Contributing 🙌🏽

Yes, I am willing to help answer this question in a PR

**Comment by DiegoGonzalezCruz** [comment]
hi! were you able to fix this?
